src/cfnlint/maintenance.py

```diff
@@ -186,3 +243,171 @@ def update_iam_policies():
+            elif t not in ['string', 'integer', 'number', 'boolean']:
+                if propdetails.get('$ref'):
+                    results.update(get_object_details(
+                        names + [propname], schema.get('definitions').get(t.get('$ref').split('/')[-1]), schema))
+                elif isinstance(t, list):
+                    LOGGER.info('Type for %s object and %s property is a list', names[0], propname)
+                else:
+                    LOGGER.info('Unable to handle %s object for %s property', names[0], propname)
+            elif t == 'string':
+                if not results.get('.'.join(names + [propname])):
+                    if propdetails.get('pattern') or (propdetails.get('minLength') and propdetails.get('maxLength')) or propdetails.get('enum'):
+                        results['.'.join(names + [propname])] = {}
+                if propdetails.get('pattern'):
+                    p = propdetails.get('pattern')
+                    if '.'.join(names + [propname]) == 'AWS::OpsWorksCM::Server.CustomPrivateKey':
+                        # one off exception to handle a weird parsing issue in python 2.7
+                        continue
+                    if sys.version_info[0] == 2:
+                        # for python 2 strings can be unicode
+                        if isinstance(p, unicode):  #pylint: disable=undefined-variable
+                            try:
+                                p = p.decode('ascii')
+                            except:  #pylint: disable=bare-except
+                                continue
+                    else:
+                        # python 3 has the ability to test isascii
+                        # python 3.7 introduces is ascii so switching to encode
+                        try:
+                            p.encode('ascii')
+                        except UnicodeEncodeError:
+                            continue
+                    try:
+                        if '\\p{' in p:
+                            continue
+                        re.compile(p, re.UNICODE)
+                        results['.'.join(names + [propname])].update({
+                            'AllowedPatternRegex': p
+                        })
+                    except:  #pylint: disable=bare-except
+                        LOGGER.info(
+                            'Unable to handle regex for type %s and property %s with regex %s', names[0], propname, p)
+                if propdetails.get('minLength') and propdetails.get('maxLength'):
+                    results['.'.join(names + [propname])].update({
+                        'StringMin': propdetails.get('minLength'),
+                        'StringMax': propdetails.get('maxLength'),
+                    })
+                if propdetails.get('enum'):
+                    results['.'.join(names + [propname])].update({
+                        'AllowedValues': propdetails.get('enum')
+                    })
```

Comment:
[Could grab `enums` for types other than `string` as well](https://json-schema.org/understanding-json-schema/reference/generic.html#enumerated-values)

---

Especially since we seem to already be handling more types in the rule itself:
https://github.com/aws-cloudformation/cfn-python-lint/blob/853522ff0e4c977331ac677a47059cf96b6f3189/src/cfnlint/rules/resources/properties/AllowedValue.py#L38